core/types/block.go

```diff
@@ -147,6 +147,16 @@ func rlpHash(x interface{}) (h common.Hash) {
 	return h
 }
 
+func typedRlpHash(typ uint8, x interface{}) (h common.Hash) {
```

Comment:
Any thoughts on this approach to concatenating the type byte at the front of the RLP data?